Makefile

```diff
@@ -0,0 +1,53 @@
+SHELL:=/bin/bash
+.PHONY: all go json md html clean 
+
+# Path Related
+MKFILE_PATH := $(abspath $(lastword $(MAKEFILE_LIST)))
+MKFILE_DIR := $(dir $(MKFILE_PATH))
+
+RELEASE?=v1alpha1
+
+# Temporary generated golang model file and directory
+TMP_GO_PACKAGE := ${MKFILE_DIR}${RELEASE}/easemesh
+TMP_DIR := ${TMP_GO_PACKAGE}/api/${RELEASE}
+
+# Target doc files 
+MD_FILENAME := meshmodel.md
+HTML_FILENAME := meshmodel.html
+GO_FILENAME := meshmodel.pb.go
+JSON_FILENAME := meshmodel.json
+
+PROTO := $(shell find ${MKFILE_DIR}${RELEASE} -type f -name "*.proto") 
+
+go: ${PROTO} 
+	@echo "generate golang file from proto: ${PROTO} "
+		@echo ""
+	protoc -I ${MKFILE_DIR}${RELEASE} --go_out=${MKFILE_DIR}${RELEASE}/ ${PROTO}  && \
```

Comment:
Where can I find the `protoc`?  You should not assume the user has installed it!

Reply:
I though this dependences should be mentioned in READM. Let me prepare it at the next PR?

Reply:
Can you refer to Makefile of others? I think that Makefile is dedicated to managing dependencies

Reply:
Actually, I referred to EG's Makefile as an example, in EG's Makefile we had made an assumption: use is working in the Golang environment. So in the mesh-api project, I thought we could just tell users how to install `protoc` in README.
Or we can use docker container cmd for using `protoc` in Makefile, but is that mean we should check and install docker for users? 
And I had searched the [Istio's API repo](https://github.com/istio/api/search?p=2&q=protoc), still can't find the `protoc` installing step...
I am confused here.

Reply:
Are you seriously or carefully check [Istio's API repo](https://github.com/istio/api), Istio has its own solution instead of `protoc`

Reply:
Already updated to check `protoc`, `protoc-gen-doc` and `protoc-gen-go` dependence.